integration-tests/pkg/executor/executor.go

```diff
@@ -0,0 +1,35 @@
+package executor
+
+import (
+	"os/exec"
+
+	"github.com/stackrox/collector/integration-tests/pkg/config"
+)
+
+type Executor interface {
+	CopyFromHost(src string, dst string) (string, error)
+	PullImage(image string) error
+	IsContainerRunning(container string) (bool, error)
+	ContainerExists(container interface{}) (bool, error)
+	ContainerID(container interface{}) string
```

Comment:
I think this makes sense, I initially went with `interface{}` because I believe the caller will always know what type of executor they are using, but if in the future we want to have a test that can be run either on docker or k8s that might not be the case anymore.

I'll rename `PodFilter` to `ContainerFilter` and change how it is used in the tests.

Reply:
Actually, the other reason I went with `interface{}` is that I could get away with not modifying every single place these methods were called, I guess that's not the case now 🤷🏻‍♂️